pkg/backend/http/proxy.go

```diff
@@ -219,8 +219,14 @@ func fallbackDialer(transport *http.Transport) func(network string, addr string)
 			transport.DialTLS = dialer
 			return conn, err
 		}
-		switch err.(type) {
-		case x509.UnknownAuthorityError:
+
```

Comment:
Just curious, why this change was needed?

Reply:
Go 1.20 introduced the [`tls.CertificateVerificationError`](https://pkg.go.dev/crypto/tls#CertificateVerificationError) and the dialling was returning that in place of `x509.UnknownAuthorityError` . In some tests I had checked with `errors.As` for the old error in case it was wrapped but seems it is not the case now.